doc/contrib/documentation.md

```diff
@@ -0,0 +1,84 @@
+# Documentation
+
+This document gives guidelines for contributors concerning Elektra's documentation.
+This document takes preference to the actual situation, so if you see documentation
+not according to this document, please [create an issue](https://issue.libelektra.org)
+or directly fix it with your next PR.
+
+> Note: It is always allowed to improve the documentation, in every PR, even if
+> the documentation fix is completely unrelated.
+
+## Target Groups
+
+We write documentation for three groups:
+
+- User:
+  Is a person who uses some tool of Elektra, e.g. `kdb`, Qt-Gui or Web-UI.
+- Developer:
+  Is a person who uses some library of Elektra within an application, tool or plugin.
+- Contributor
+  Is a person who does changes within Elektra's repository.
+
+There are 3 separate folders for these 3 groups:
+
+- [for users](/doc/help)
+- [for developers](/doc/dev)
+- [for contributors](/doc/contrib)
+
+> Takeaway:
+> Every document must have a clear target group (user, developer or contributor).
+> Sometimes it is clear from the directory sometimes it must be explicitly stated,
+> e.g. tutorials.
+
+## Orientation
+
+Each documentation should clearly be oriented to one of these 3 directions:
+
+1. learning-oriented: first introduction is done via tutorials, it teaches the basics.
+2. information-oriented: the README.md and API docs together are the references: they cover everything that someone needs to know about a [module](/doc/help/elektra-glossary.md).
+3. understanding-oriented: the [doc/decisions](/doc/decisions) explain the "Why?" something is done as it is done (Rationale).
+
+In literature there is also goal-oriented but we prefer learning-oriented approaches.
+E.g. of course you might have the goal to write a new plugin.
+But why not also learn about plugins while creating a new plugin?
+
+> Takeaway:
+> Don't try to combine different orientations in one document,
+> instead spit your documentation up in e.g. a README.md (information),
+> tutorial (learning),
+> decisions (understanding).
+
+## Criteria
+
+Elektra's documentation must fulfill:
+
+- It is self-contained.
+  It is not enough to link to some paper and an external page to explain, e.g., decisions.
+  All explanation must be within the repo, e.g., in case the external information goes away.
```

Comment:
Should mention that linking to external sources for further details is fine, as long as our docs explain enough that everything is clear if the external source goes away.

An exception might also if the external source is authoritative. For example I'd say `toml` should link to  https://toml.io/en/v1.0.0 and state something like "We follow the spec with these exceptions: ...". It would be useless to copy the entire spec into our docs. That would just mean people need to read both versions to find potential differences.